### DISCUSSION TITLE
Allure object restart after codeceptjs async action

### DISCUSSION THREAD

**Discussion by shemeshsapir** [allure-framework/allure2#2710] [Questions & Support] [answered]
Hi, I'm using the following packages in my project.:
    "allure-codeceptjs": "^2.15.1",
    "allure-commandline": "^2.13.0",
    
I have a problem that codeceptjs async action like - 
"await I.grabCurrentUrl()" for example, make the allure object to be null (The "currentTest" field is null, etc..).

this is the code (important to say that I tried also to create allure instance inside the eventListener and sent it as argument to the method):

event.dispatcher.on(event.test.after, async (test) => {
      console.log(`Start event listener "after test" with test: ${test.title}`);
      const email = I.getLoggedInEmail();
      const url = await I.grabCurrentUrl(); **// This is the problem **
      createAllureReport(email, url);
});
   
   
  "createAllureReport" implementation:
  
   export const createAllureReport = (email: string, url: string): void => {
  try {
    allure = getAllureInstance();
    allure.issue("Link to url:", url);
    allure.parameter("Email:", email);
  } catch (error) {
    output.error(`Failed to create Allure report: ${error}`);
  }
  
  and "getAllureInstance" implementation:
  
  const getAllureInstance = () => {
  if (!allure) {
    allure = codeceptjs.container.plugins("allure");
    console.log("Allure instance created.");
  }
  return allure;
};

any explanation/ideas why? and hoe to solve it?

**Comment by baev** [comment, marked as answer]
Please, try to use the latest beta version available, the CodeceptJS integration was reworked and now it should work correctly.

**Reply by shemeshsapir** [reply]
Is it safe to use beta version?

**Reply by baev** [reply]
Yes, it's stable. There are no more changes planned to `allure-codeceptjs` before the 3.0.0 release unless some of the users report some issues with it

**Reply by shemeshsapir** [reply]
I will try - thank you!

**Reply by shemeshsapir** [reply]
Hi, I tried the last version and it's much worse for me
because it's not supporting 'after' events.
In addition, I can't see labels that I added even in the 'test before' event listener :(

I'm getting this error:
metadata messages isn't supported for after test fixtures

**Reply by shemeshsapir** [reply]
?

**Reply by baev** [reply]
Please create a separate discussion per issue you have
